components/eamxx/src/diagnostics/pbl_entrainment_budget.cpp

```diff
@@ -0,0 +1,332 @@
+  auto o_qt_caret = ekat::subview_1(out, m_index_map["qt^"]);
+  auto o_qt_ttend = ekat::subview_1(out, m_index_map["qt_ttend"]);
+  auto o_df_inpbl = ekat::subview_1(out, m_index_map["dF"]);
+
+  // Get the input views
+  const auto &qc_v = get_field_in("qc").get_view<Real **>();
+  const auto &qv_v = get_field_in("qv").get_view<Real **>();
+  const auto &tm_v = get_field_in("T_mid").get_view<Real **>();
+  const auto &pm_v = get_field_in("p_mid").get_view<Real **>();
+  const auto &pd_v = get_field_in("pseudo_density").get_view<Real **>();
+  const auto &sd_v = get_field_in("SW_flux_dn").get_view<Real **>();
+  const auto &su_v = get_field_in("SW_flux_dn").get_view<Real **>();
+  const auto &ld_v = get_field_in("LW_flux_dn").get_view<Real **>();
+  const auto &lu_v = get_field_in("LW_flux_up").get_view<Real **>();
+
+  // tracked stuff
+  const auto &prev_qtot_v = m_prev_qt.get_view<Real **>();
+  const auto &prev_tliq_v = m_prev_tl.get_view<Real **>();
+
+  view_2d qt_v("qt_v", m_ncols, m_nlevs);
+  view_2d tl_v("tl_v", m_ncols, m_nlevs);
+  calc_tl_qt(tm_v, pm_v, qv_v, qc_v, tl_v, qt_v);
+
+  const auto &curr_ts =
+      get_field_in("qc").get_header().get_tracking().get_time_stamp();
+
+  auto dt = curr_ts - m_start_t;
+
+  const int num_levs  = m_nlevs;
+  const int pblinvalg = m_pblinvalg;
+  const auto policy   = ESU::get_default_team_policy(m_ncols, m_nlevs);
+
+  constexpr int wsms = 1;
+  using WSMgr        = ekat::WorkspaceManager<Real, DefaultDevice>;
+  WSMgr wsm(num_levs, wsms, policy);
+
+  Kokkos::parallel_for(
+      "Compute " + name(), policy, KOKKOS_LAMBDA(const MT &team) {
+        const int icol = team.league_rank();
+        // inputs
+        const auto qc_icol = ekat::subview(qc_v, icol);
+        const auto qv_icol = ekat::subview(qv_v, icol);
+        const auto tm_icol = ekat::subview(tm_v, icol);
+        const auto pm_icol = ekat::subview(pm_v, icol);
+        const auto pd_icol = ekat::subview(pd_v, icol);
+        const auto sd_icol = ekat::subview(sd_v, icol);
+        const auto su_icol = ekat::subview(su_v, icol);
+        const auto ld_icol = ekat::subview(ld_v, icol);
+        const auto lu_icol = ekat::subview(lu_v, icol);
+
+        // tracked
+        const auto qt_icol = ekat::subview(qt_v, icol);
+        const auto tl_icol = ekat::subview(tl_v, icol);
+
+        auto prev_qtot_icol = ekat::subview(prev_qtot_v, icol);
+        auto prev_tliq_icol = ekat::subview(prev_tliq_v, icol);
+
+        auto ws = wsm.get_workspace(team);
+        ekat::Unmanaged<WSMgr::view_1d<Real>> tm_grad;
+        ws.take_many_contiguous_unsafe<wsms>({"tm_grad"}, {&tm_grad});
+
+        // We first want to find the PBL inversion. There are three methods to
+        // do so. All our methods here rely on the *gradient* of state fields
+        // (for now). First, we can simply find the first level from the surface
+        // that has a a temperature "inversion" (temperature goes positive
```

Comment:
do you mean "temperature _gradient_ inversion"? Cause a negative T would be a bit concerning...

Reply:
Looking at the code below, you are looking for the minimum gradient. From your comment, I thought you wanted to find the place where the inversion _starts_. Is the comment misleading? Or is the code wrong?